src/buffer.rs

```diff
@@ -0,0 +1,301 @@
+use crate::channel::mpsc::Sender as PrioritySender;
+use futures::channel::mpsc::Sender;
+use std::{
+    collections::VecDeque,
+    task::{Context, Poll},
+};
+
+/// If the buffer capacity is greater than u8 max, shrink it
```

Comment:
The comment is misleading.

```suggestion
/// If the buffer unused capacity is greater than u8 max, shrink it
```